src/core/lib/attribute-manager.js

```diff
@@ -513,16 +520,30 @@ export default class AttributeManager {
   // Update attribute buffers from any attributes in props
   // Detach any previously set buffers, marking all
   // Attributes for auto allocation
-  /* eslint-disable max-statements */
+  /* eslint-disable max-statements, no-continue */
   _setExternalBuffers(bufferMap) {
     const {attributes, numInstances} = this;
 
     // Copy the refs of any supplied buffers in the props
     for (const attributeName in attributes) {
       const attribute = attributes[attributeName];
       const buffer = bufferMap[attributeName];
-      attribute.isExternalBuffer = false;
-      if (buffer) {
+
+      if (!buffer) {
+        attribute.isExternalBuffer = false;
+        continue;
+      }
+
+      attribute.isExternalBuffer = true;
+      attribute.needsUpdate = false;
+
+      if (buffer instanceof Buffer) {
+        attribute.value = null;
```

Comment:
Should we check or assert if Buffer is not big enough compared to `numInstances` ?
Also should we overwrite other values in attribute, like size, type with of those from Buffer ?